lib/block.ml

```diff
@@ -300,7 +300,7 @@ let guess_ocaml_kind contents =
     | h :: t ->
         let h = String.trim h in
         if h = "" then aux t
-        else if String.length h > 1 && h.[0] = '#' then `Toplevel
+        else if String.length h > 2 && h.[0] = '#' && h.[1] = ' ' then `Toplevel
```

Comment:
If needed, the check could be more strict as top-level directives necessarily start with a lowercase ASCII letter (or the sequence `\#` which is never going to be used).